File: compiler/cpp/src/exception/UnexpectedCharException.cpp
/*
Copyright © 2021-2022 Compose Generator Contributors
All rights reserved.
*/

/*
Copyright © 2021 Compose Generator Contributors
All rights reserved.
*/

#include "UnexpectedCharException.h"

const char *UnexpectedCharException::what() const noexcept {
    return errorMessage.c_str();
}

File: compiler/java/src/main/java/com/chillibits/ccom/reader/UnexpectedCharException.java
/*
Copyright © 2021 Compose Generator Contributors
All rights reserved.
*/

package com.chillibits.ccom.reader;

/**
 * Exception indicating that the next char in the file did not equal the expected char.
 */
public class UnexpectedCharException extends Exception {

    public UnexpectedCharException(char expected, char got, String position) {
        super("Expected '" + expected + "' but got '" + got + "' " + position);
    }

}
